service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/ERLookupController.java

```diff
@@ -0,0 +1,90 @@
+
+package com.ericsson.eiffel.remrem.generate.controller;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class ERLookupController {
+    private static RestTemplate restTemplate = new RestTemplate();
+
+    @Autowired
+    public void setRestTemplate(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public static ArrayList<String> ErlookupJson(JsonObject bodyJson) {
+        String[] ids = null;
+        ArrayList<String> aList = new ArrayList<String>();
+        JsonArray links = bodyJson.get("eventParams").getAsJsonObject().get("links").getAsJsonArray();
+        String s = links.toString();
+        JsonParser parser = new JsonParser();
+        JsonArray elemArray = parser.parse(s).getAsJsonArray();
+        for (int i = 0; i < elemArray.size(); i++) {
```

Comment:
These for loops can broken up in functions and better unit tested.

Reply:
ok.. I am currently working on unit and functional testcases.

Reply:
ok.